tests/test_data_snapshot.py

```
@@ -0,0 +1,56 @@
+
+    dest = snapshot_data_dir(data)
+
+    backed = dest / "leak"
+    assert backed.is_symlink()                      # preserved as a link
+    assert not backed.read_text() == "SENSITIVE" or backed.is_symlink()
+    # the symlink target's content was not materialized as a real file
+    assert os.path.islink(backed)
+
+
+def test_backup_dir_is_owner_only(tmp_path):
+    data = tmp_path / "data"; data.mkdir()
```

Ruff (0.15.15)

[error] Multiple statements on one line (semicolon)

(E702)